record_generator.go

```diff
@@ -0,0 +1,119 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package generator
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/google/uuid"
+)
+
+type recordGenerator interface {
+	generate() (sdk.Record, error)
+}
```

Comment:
I'll just say YAGNI. My personal opinion is that we should strive to keep the code simple and not introduce abstractions that have no value and we're not sure if we need it. When the need arises we can add it, we'll have a better understanding what we need at that moment anyway.

Reply:
> My personal opinion is that we should strive to keep the code simple and not introduce abstractions that have no value and we're not sure if we need it.

I completely agree with you on this!

> When the need arises we can add it, we'll have a better understanding what we need at that moment anyway.

The need here is making the source's responsibility that of controlling the overall flow, and making the record generator's responsibility just that. That way, I can write tests for source which focus just on the flow. For example, right now, when testing the read times, or the sleep and generate cycles, I also have to specify format.type and format.options, even though I don't care about it. 

On the other hand, if I want to test the record generation, I need to set up a source.

Reply:
> I completely agree with you on this!

Good to hear we're on the same page here! I understand why we would want to introduce such an interface, all I'm saying is that it's unnecessary code right now and serves no need **right now**. I'm not saying the idea doesn't make sense, I'm saying that it's not the right time to introduce an interface if it's not being used. This is a path towards [interface pollution](https://rakyll.org/interface-pollution/).

I won't insist on this, let's leave it as it is, all I ask is to please be mindful of this in the future.

Reply:
FTR, we discussed this offline. Even though we could use the interface (or any separate type) to have better tests (it would help mocking), our focus is not on this connector (as it's used in tests and demos only), so we don't need a sophisticated testing infrastructure.